src/sage/repl/preparse.py

```diff
@@ -1253,6 +1263,10 @@ def preparse_numeric_literals(code, extract=False, quotes="'"):
         'RealNumber(str().join(map(chr, [51, 46, 49, 52])))'
         sage: preparse_numeric_literals('5j', quotes=None)
         'ComplexNumber(0, str().join(map(chr, [53])))'
+        sage: preparse_numeric_literals("1" * 4301, quotes=None) == f'Integer(str().join(map(chr, {[49] * 4301})))'
+        True
+        sage: preparse_numeric_literals("1" * 4301, quotes="'''") == f"Integer('''{"1" * 4301}''')"
```

Comment:
```suggestion
        sage: preparse_numeric_literals("1" * 4301, quotes="'''") == f"Integer('''{'1' * 4301}''')"
```
I think this test has become a bit overkill now. I would consider removing it.

Reply:
Fixed! Althought my local test using `sage -t .../preparse.py` succeeded with double quotes inside double quotes 🤔, but still good to know, thanks!